target_chains/ethereum/contracts/contracts/pyth/Pyth.sol

```diff
@@ -89,6 +88,8 @@ abstract contract Pyth is
                 i++;
             }
         }
+        uint requiredFee = getRequiredFee(totalNumUpdates, updateData.length);
```

Comment:
how about this instead?
```
 if (totalNumUpdates == 0){
          totalNumUpdates = updateData.length;
 }
 requiredFee = getTotalFee(totalNumUpdates);
```
the signatures between `getFeeAccumulator` & `getFeeBatch` are the same so for saving on bytesize maybe it would be best to not have 2 separate `getFeeXXX` methods?